### PR TITLE
Speed up indices stats yaml tests

### LINKED ISSUE #95580: Reduce the time it takes to run the field data stats yaml tests
Reduce the time it takes to run the yaml rest tests introduced by this change: #94500

These tests are very slow: https://gradle-enterprise.elastic.co/s/mjewt7di232pk/tests/overview?class=org.elasticsearch.xpack.security.CoreWithSecurityClientYamlTestSuiteIT

### PR DESCRIPTION
**Pull request by martijnvg**
The tests in `13_fields.yml` file are taking a very long time in any test cluster with a single node. This is because the create index api call in the test setup waits for all shards to be allocated. Also for the replica shards and that never happens in a single node cluster. The create index api just waits 30 seconds for this to happen before sending back a response. Adding at least 30 seconds of execution time to each test in this yaml file.

This seems to be added a while ago when dealing with mixed cluster rest test failures. No shards seem to be available in that past causing test failures. But the subsequent cluster health api should be sufficient to ensure that at least one copy per shard is allocated.

This change also changes the number of shards from 3 to 2 (2 shards is sufficient for this test), targets to cluster health api call only to the `test1` index and splits the global ordinal based tests into its own file. The reason for the latter is that testing global ordinal stats has its own requirements that the other tests don't need (the need of an extra segment, running actual aggregations, but for these tests 1 shard is fine).

Closes #95580

### CODE REVIEW

**Comment by martijnvg** [comment]
These tests no longer seem to be slow: https://gradle-enterprise.elastic.co/s/v4422c5psxc2s/tests/overview
compared to test run earlier today: https://gradle-enterprise.elastic.co/s/ctastauwxhbvs/tests/overview

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.stats/13_fields.yml [hunk 1]
```diff
@@ -4,13 +4,12 @@ setup:
...
       indices.create:
           index:  test1
-          wait_for_active_shards: all
           body:
               settings:
                 # Limit the number of shards so that shards are unlikely
                 # to be relocated or being initialized between the test
                 # set up and the test execution
-                index.number_of_shards: 3
+                index.number_of_shards: 2
```
Comment (kkrik-es) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.stats/13_fields.yml:12]: Nit: do we even need to set this?

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.stats/90_global_ordinals.yml [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+      index:
+        index: test1
+        id: "2"
+        body: { "bar": "foo", "baz": "foo" }
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      search:
```
Comment (kkrik-es) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.stats/90_global_ordinals.yml:44]: Move this to a test?

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.stats/13_fields.yml [hunk 3]
```diff
@@ -107,23 +84,6 @@ setup:
...
-
-  - do:
-      indices.stats: {}
-
-  - match: { _shards.failed: 0}
-  - gt:       { _all.total.fielddata.memory_size_in_bytes: 0 }
-  - is_false:   _all.total.fielddata.fields
-  - gte:       { _all.total.fielddata.global_ordinals.build_time_in_millis: 0 }
-  - is_false:   _all.total.fielddata.global_ordinals.fields
-  - gt:       { _all.total.completion.size_in_bytes: 0 }
```
Comment (kkrik-es) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.stats/13_fields.yml:124]: These are dropped below, just verifying that it's intended.

**Review by kkrik-es — APPROVED** [review]
LGTM, just a few minor comments.

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.stats/13_fields.yml (hunk 1 above)
Comment (martijnvg) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.stats/13_fields.yml:12]: Many of these stats are per shard. I think the intention is to verify that these tests work with multiple shards. If it works with multiple shards it also works when indices stats targets an index with a single shard.

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.stats/13_fields.yml (hunk 3 above)
Comment (martijnvg) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.stats/13_fields.yml:124]: I forgot to mention this. Yes, this is intended, there was duplication between tests.

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.stats/90_global_ordinals.yml (hunk 2 above)
Comment (martijnvg) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/indices.stats/90_global_ordinals.yml:44]: This isn't really a test. It is needed for the indices stats to be able to report global ordinals. In this case they are computed lazily when needed by a search with aggs.

**Review by csoulios — APPROVED** [review]
LGTM

**Comment by mark-vieira** [comment]
@martijnvg I think this should be backported to 8.7 as well. Seems we have the same behavior there: https://gradle-enterprise.elastic.co/s/4idduojrcr7so/tests/overview?class=org.elasticsearch.xpack.security.CoreWithSecurityClientYamlTestSuiteIT

**Comment by martijnvg** [comment]
@mark-vieira Good point. I've opened a backport pr: #95751
 I think this should even be backported even to the 7.17 branch.

**Pull request merged**